### PR TITLE
[pro#363-3] General changes

### PR DESCRIPTION
**Pull request by gbp**
Fixes the following part of https://github.com/mysociety/alaveteli-professional/issues/363

> - [x] Probably replace OpenStruct with stub in XWithY decorator spec

Along with a few more issues I identified

1. Default styling of `a.button:visited` links appears as blue on blue
2. Returning an 404 when going to `/profile/subscriptions` seems the unnecessary confusing, so now redirects to `/plana/pro` if users doesn't have an active subscription

### CODE REVIEW

File: spec/models/alaveteli_pro/subscription_with_discount_spec.rb [hunk 1]
```diff
@@ -22,14 +22,14 @@ def mock_subscription(coupon)
     end
 
     it 'applies a percentage discount correctly' do
-      coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: true)
+      coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: true)
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply spec/models/alaveteli_pro/subscription_with_discount_spec.rb]: probably worth making Hound happy here as it's not going to be too painful

File: spec/models/alaveteli_pro/subscription_with_discount_spec.rb [hunk 2]
```diff
@@ -50,7 +50,7 @@ def mock_subscription(coupon)
     context 'the discount is invalid' do
 
       it 'returns false' do
-        coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: false)
+        coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: false)
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply spec/models/alaveteli_pro/subscription_with_discount_spec.rb]: probably worth making Hound happy (as above)

File: spec/models/alaveteli_pro/subscription_with_discount_spec.rb [hunk 3]
```diff
@@ -61,7 +61,7 @@ def mock_subscription(coupon)
     context 'a valid discount applies' do
 
       it 'returns true' do
-        coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: true)
+        coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: true)
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply spec/models/alaveteli_pro/subscription_with_discount_spec.rb]: probably worth making Hound happy (as above)

File: spec/models/alaveteli_pro/subscription_with_discount_spec.rb [hunk 4]
```diff
@@ -76,7 +76,7 @@ def mock_subscription(coupon)
     context 'the price is > 0' do
 
       it 'returns false' do
-        coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: true)
+        coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: true)
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply spec/models/alaveteli_pro/subscription_with_discount_spec.rb]: probably worth making Hound happy (as above)

File: spec/models/alaveteli_pro/subscription_with_discount_spec.rb [hunk 5]
```diff
@@ -87,7 +87,7 @@ def mock_subscription(coupon)
     context 'there is a 100% discount' do
 
       it 'returns true' do
-        coupon = OpenStruct.new(id: "100_off", percent_off: 100, valid: true)
+        coupon = double(:coupon, id: "100_off", amount_off: nil, percent_off: 100, valid: true)
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply spec/models/alaveteli_pro/subscription_with_discount_spec.rb]: probably worth making Hound happy (as above) - sorry, these are basically my fault!

File: app/controllers/alaveteli_pro/base_controller.rb [hunk 6]
```diff
@@ -16,9 +16,9 @@ def pro_user_authenticated?(reason_params = nil)
     if reason_params.nil?
       reason_params = {
         web: _("To access {{pro_site_name}}",
-                pro_site_name: AlaveteliConfiguration.pro_site_name),
+               pro_site_name: AlaveteliConfiguration.pro_site_name),
         email: _("Then you can access {{pro_site_name}}",
-                pro_site_name: AlaveteliConfiguration.pro_site_name)
+               pro_site_name: AlaveteliConfiguration.pro_site_name)
```
[the remark this answers is not in the document]
Comment (lizconlan) [reply app/controllers/alaveteli_pro/base_controller.rb]: wrong way! should align under the double quote

File: app/controllers/alaveteli_pro/subscriptions_controller.rb [hunk 7]
```diff
@@ -124,4 +123,11 @@ def authenticate
     authenticated?(post_redirect_params)
   end
 
+  def check_existing_subscriptions
+    # TODO: This doesn't take the plan in to account
+    unless @user.pro_account.try(:active?)
+      flash[:notice] = _('You aren\'t currently subscribed to any plans')
```
Comment (lizconlan) [inline app/controllers/alaveteli_pro/subscriptions_controller.rb]: maybe the message should be `You don't currently have an active Pro subscription` as the terminology around plans isn't directly exposed to users - we can expand it to include whatever we call Batch when we get to it

**Comment by gbp** [comment]
@lizconlan I've rebased onto develop as most of those hound warnings have already been dealt with on a different branch. Added a couple of fixups for the remaining issues.

**Pull request closed without merging**